syntax = "proto3";

message Request {
  string robot_name = 1;
  string request_name = 2;
  string station_name = 3;
  string cart_name = 4;
  string table_name = 5;
  string job_name = 6;
  string rdb_data = 7;
  string job_status = 8;
}

message Row {
  int32 row_identifier = 1;
  string column_values = 2;
}

message TableData {
  string table_name = 1;
  repeated Row rows = 2;
}

message Response_UpdateRDB {
  repeated TableData tables = 1;
}

message Response_PullLDB {
  bytes ldb = 1;
}

message Response_Job {
    string job_id = 1;
    string job_type = 2;
    string robot_name = 3;
    string cart = 4;
    string source_station = 5;
    string target_station = 6;
}

message Response_FetchJob {
    string message = 1;
    Response_Job job = 2;

}

message Response_FreeStation {
  string station_name = 1;
}

message Response_PushToLDB {
  bool success = 1;

}

message Response_ResetStationBlocker {
  bool success = 1;

}

message Response_UpdateJobMonitor {
  bool success = 1;

}

message Response_OperationTime {
  int64 msec = 1;

}

message Response_Ready2PlugInADS {
  bool ready_to_plugin = 1;

}

service Communication {
  rpc UpdateRDB(Request) returns (Response_UpdateRDB);
  rpc PullLDB(Request) returns (Response_PullLDB);
  rpc UpdateJobMonitor(Request) returns (Response_UpdateJobMonitor);
  rpc FetchJob(Request) returns (Response_FetchJob);
  rpc AskFreeStation(Request) returns (Response_FreeStation);
  rpc PushToLDB(Request) returns (Response_PushToLDB);
  rpc ResetStationBlocker(Request) returns (Response_ResetStationBlocker);
  rpc OperationTime(Request) returns (Response_OperationTime);
  rpc Ready2PlugInADS(Request) returns (Response_Ready2PlugInADS);

}
